include/gridtools/storage/adapter/nanobind_adapter.hpp

```diff
@@ -25,24 +25,24 @@
 namespace gridtools {
     namespace nanobind_sid_adapter_impl_ {
 
-        // Use nanobind::any for dynamic stride, use an integral value for static stride.
-        template <std::size_t... Values>
-        using stride_spec = std::index_sequence<Values...>;
+        // Use `-1` for dynamic stride, use an integral value for static stride.
+        template <ssize_t... Values>
```

Comment:
I think we use `int` for indices internally at quite a few places in GridTools, so maybe this would also be good enough here (this might even have a performance impact).

Reply:
I actually used the wrong `ssize_t`. It's actually a nb type defined here https://github.com/wjakob/nanobind/blob/c5ae2a36a704c1c62da99a81fad919261a3e9848/include/nanobind/nb_traits.h#L11.

I'll use that. We could still convert (runtime) strides to `int`, but I would do separately and probably should look at performance.